src/shared/options.ts

```diff
@@ -119,7 +119,12 @@ export class Options {
             'omnisharp.useEditorFormattingSettings',
             true
         );
-        const enableRoslynAnalyzers = Options.readOption<boolean>(config, 'omnisharp.enableRoslynAnalyzers', false);
+        const diagnosticAnalysisScope = Options.readOption<string>(
+            config,
+            'dotnet.backgroundAnalysis.analyzerDiagnosticsScope',
+            useOmnisharpServer ? 'none' : 'openFiles'
+        );
+        const enableRoslynAnalyzers = diagnosticAnalysisScope != 'none';
```

Comment:
This looks good but I think we need to support the case where someone has specified a custom value for O# in their settings.json.  We probably still need to read the O# option names (even though they aren't defined in the package.json) and convert them to the new option.  

For other options that have the same value this is easier (see an example on line 52), but for this one we'll have to have some custom code to convert the values.

Not required for this PR as well - but we also migrate the old O# option names to the new dotnet names in here - https://github.com/dotnet/vscode-csharp/blob/main/src/shared/migrateOptions.ts#L9
Again it might be a bit more complicated for this option since we need to convert two options into one, but something to consider.

Reply:
Thanks David. As you mentioned, given that the new and legacy options are different data types, its a bit more trickier to handle the migration. I have added back the logic to respect the legacy options, though my only concern is that if that forces us back to have the UX for the same. For example, if user had enabled analyzers with legacy option, and now they want to disable it, we only allow them to change the new option value and because we implicitly respect the old option value, they are stuck with not being able to disable analyzers.

Reply:
Yeah - lets take this, and at some point later on we can update the migration code to migrate the old option values to the new ones so we can avoid that issue.

Reply:
Filed https://github.com/dotnet/vscode-csharp/issues/6090 to track this work